source/s3_request_messages.c

```diff
@@ -586,11 +635,22 @@ struct aws_http_message *aws_s3_abort_multipart_upload_message_new(
     return NULL;
 }
 
+static int s_is_get_request(const struct aws_http_message *message, bool *is_request) {
+    struct aws_byte_cursor method;
+    if (aws_http_message_get_request_method(message, &method)) {
```

Comment:
feel like this should not fail, maybe an ASSERT on it, so, you don't need a return code?